src/compressed_tensors/transform/utils/matrix.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, Optional, Tuple
+
+import torch
+from compressed_tensors.transform import TransformLocation
+
+
+__all__ = ["get_transform_size", "apply_transform_weight"]
+
+
+def get_transform_size(
+    module: torch.nn.Module,
+    location: TransformLocation,
+    head_dim: Optional[int] = None,
+) -> int:
+    """
+    Determine the size of a transform matrix given its location on the module
+
+    :param module: module that matrix will be applied to
+    :param location: location on module
+    :param head_dim: size of head when transform is applied to mha
+    :return: size of matrix
+    """
+    if isinstance(module, torch.nn.Linear):
+        if location in (TransformLocation.INPUT, TransformLocation.WEIGHT_INPUT):
+            size = module.in_features
+        else:
+            size = module.out_features
+    else:
```

Comment:
don't we also need torch.nn.Embedding here?

```suggestion
    elif isinstance(module, torch.nn.Embedding):
        if location in (TransformLocation.INPUT, TransformLocation.WEIGHT_INPUT):
            return module.num_embeddings
        else:
            return module.embedding_dim
    else:
```